src/main/java/com/github/bibsysdev/nva/cristin/person/CristinApiClient.java

```diff
@@ -0,0 +1,190 @@
+
+import static com.github.bibsysdev.nva.cristin.person.UriUtils.buildUri;
+import static java.util.Arrays.asList;
+import static nva.commons.core.attempt.Try.attempt;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.bibsysdev.nva.cristin.person.model.cristin.CristinPerson;
+import com.github.bibsysdev.nva.cristin.person.model.nva.NvaPerson;
+import com.github.bibsysdev.nva.cristin.person.model.nva.EmptyNvaPerson;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.JsonUtils;
+import nva.commons.core.attempt.Failure;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@JacocoGenerated
+public class CristinApiClient {
+
+    protected static final String PERSON_LOOKUP_CONTEXT_URL = "https://example.org/person-context.json";
+    private static final Logger logger = LoggerFactory.getLogger(CristinApiClient.class);
+    private static final Environment ENVIRONMENT = new Environment();
+    protected static final String BASE_URL = ENVIRONMENT.readEnv("BASE_URL");
+    private static final String CRISTIN_API_HOST = ENVIRONMENT.readEnv("CRISTIN_API_HOST");
+    private static final ObjectMapper OBJECT_MAPPER = JsonUtils.objectMapper;
+
+    private static final String NAME = "name";
+    private static final String CHARACTER_EQUALS = "=";
+    private static final String HTTPS = "https";
+    private static final String CRISTIN_API_PERSONS_PATH = "/v2/persons/";
+    private static final String SEARCH_PATH = "search?QUERY_PARAMS"; // TODO: NP-2412: Replace QUERY_PARAMS
+    private static final String ERROR_MESSAGE_FETCHING_CRISTIN_PERSON_WITH_ID = "Error fetching cristin person with "
+        + "id: %s . Exception Message: %s";
+    private static final String ERROR_MESSAGE_BACKEND_FETCH_FAILED = "Your request cannot be processed at this time "
+        + "due to an upstream error";
+
+    /**
+     * Creates a wrapper object containing CristinPersons transformed to NvaPersons. Is used for serialization to the
+     * client.
+     *
+     * @param parameters The query params
+     * @param language   Language used for some properties in Cristin API response
+     * @return a PersonsWrapper with transformed CristinPersons
+     * @throws IOException        for connection errors
+     * @throws URISyntaxException if URI is malformed
+     */
+    public PersonsWrapper queryCristinPersonsIntoWrapperObject(Map<String, String> parameters, String language)
+        throws IOException, URISyntaxException {
+
+        long startRequestTime = System.currentTimeMillis();
+        List<CristinPerson> enrichedPersons = queryAndEnrichPersons(parameters, language);
```

Comment:
you collect a stream here only to open a stream later in line 72. Move this definition closer to its use and avoid collecting and opening a stream if there is no reason for that.

Reply:
This pattern was inherited from nva-cristin-project, I believe it's the separate the cristin-api call (and time it) from the general workings of our lambda. If there's something specific you'd like me to change I'll comply, but otherwise I think this might be better suited as future work for the two projects.